sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/BinaryDataDeserializer.java

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.jackson;
+
+import com.azure.core.implementation.StringBuilderWriter;
+import com.azure.core.util.BinaryData;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import java.io.IOException;
+import java.io.Writer;
+
+/**
+ * Custom deserializer for deserializing {@link BinaryData}.
+ */
+final class BinaryDataDeserializer extends JsonDeserializer<BinaryData> {
+    /**
+     * Gets a module wrapping this serializer as an adapter for the Jackson ObjectMapper.
+     *
+     * @return a simple module to be plugged onto Jackson ObjectMapper.
+     */
+    public static SimpleModule getModule() {
+        SimpleModule module = new SimpleModule();
+        module.addDeserializer(BinaryData.class, new BinaryDataDeserializer());
+        return module;
+    }
+
+    @Override
+    public BinaryData deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+        JsonToken token = p.currentToken();
+
+        if (token == JsonToken.VALUE_NULL) {
+            return null;
+        } else if (token.isStructStart()) {
+            return BinaryData.fromString(bufferStruct(p));
+        } else {
+            return BinaryData.fromString(p.getText());
+        }
+    }
+
+    private static String bufferStruct(JsonParser parser) throws IOException {
+        StringBuilder buffer = new StringBuilder(128);
+        Writer bufferWriter = new StringBuilderWriter(buffer);
```

Comment:
Should this be closed? If not, can we just use the StringBuilder directly?

Reply:
Shouldn't need to be closed as `StringBuilderWriter` is akin to other I/O types like `ByteArrayOutputStream` where it is just a simple wrapper around a store that handles the resource.